File: python/list_credentials.py
#!/usr/bin/env python3
"""List or fetch SIP credentials for a trunk. Optional CREDENTIAL_ID for single row."""
from _client import get
import os

trunk = os.environ.get("TRUNK_SID")
if not trunk:
    print("Error: TRUNK_SID required")
    exit(1)

q = {}
if os.getenv("PAGE_SIZE"):
    q["page_size"] = os.getenv("PAGE_SIZE")
if os.getenv("PAGE_OFFSET"):
    q["offset"] = os.getenv("PAGE_OFFSET")
if os.getenv("CREDENTIAL_ID"):
    q["id"] = os.getenv("CREDENTIAL_ID")

print("Listing credentials...")
get(f"/trunks/{trunk}/credentials", query=q or None)

File: java/ListCredentials.java
public class ListCredentials {
    public static void main(String[] args) throws Exception {
        String trunk = System.getenv("TRUNK_SID");
        if (trunk == null || trunk.isEmpty()) {
            System.err.println("Error: TRUNK_SID required");
            System.exit(1);
        }
        StringBuilder q = new StringBuilder();
        String ps = System.getenv("PAGE_SIZE");
        String off = System.getenv("PAGE_OFFSET");
        String id = System.getenv("CREDENTIAL_ID");
        if (ps != null && !ps.isEmpty()) q.append("page_size=").append(ps).append("&");
        if (off != null && !off.isEmpty()) q.append("offset=").append(off).append("&");
        if (id != null && !id.isEmpty()) q.append("id=").append(id).append("&");
        String qs = q.length() > 0 ? q.substring(0, q.length() - 1) : "";
        System.out.println("Listing credentials...");
        _Client.get("/trunks/" + trunk + "/credentials", qs);
    }
}
